fabric-events-lifecycle-v0/src/main/java/net/fabricmc/fabric/impl/event/lifecycle/client/LegacyClientEventInvokers.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.event.lifecycle.client;
+
+import net.fabricmc.api.ClientModInitializer;
+import net.fabricmc.fabric.api.event.client.ClientTickCallback;
+import net.fabricmc.fabric.api.event.client.ItemTooltipCallback;
+import net.fabricmc.fabric.api.event.world.WorldTickCallback;
+import net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientLifecycleEvents;
+
+public class LegacyClientEventInvokers implements ClientModInitializer {
+	@Override
+	public void onInitializeClient() {
+		// Allows deprecated events to still be invoked by the newer implementations
+		ClientLifecycleEvents.CLIENT_TICK.register(ClientTickCallback.EVENT.invoker()::tick);
+		ClientLifecycleEvents.WORLD_TICK.register(WorldTickCallback.EVENT.invoker()::tick);
+		// This is part of generic events now.
+		net.fabricmc.fabric.api.client.event.generic.v1.item.ItemTooltipCallback.EVENT.register(ItemTooltipCallback.EVENT.invoker()::getTooltip);
```

Comment:
The invokers of all of the above legacy events are being captured here, which is invalid. That is, for each event here, the value of `EVENT.invoker()` is only evaluated in this method, where the callback is being registered for the replacement event, and not each time the new event is invoked. The Event class allows the value of `invoker()` to change when new listeners are registered.

For the array-based events which all of fabric API uses, the invoker is updated each time a new listener is registered, which means this will fail to notify any listeners that are registered after this initializer is called.

This will have to be changed to evaluate the value of `invoker()` each time the event is called, probably with an explicit lambda instead of a method reference.

Reply:
Changed it to a lambda.